com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -393,6 +405,74 @@ private void OnDestroy()
             }
         }
 
+        private SnapshotDespawnCommand GetDespawnCommand()
+        {
+            SnapshotDespawnCommand command;
+            command.NetworkObjectId = NetworkObjectId;
+            command.TickWritten = default; // will be reset in Despawn
```

Comment:
Not sure if I understand this comment. Will this value later be set correctly by the SnapshotSystem or?

Reply:
Yes, tickWritten is meant to be set by SnapshotSystem, not by the calling code. But for some obscure C# design reason, apparently that struct can't have either default constructor nor static initializer.

I'll change the comment to `// will be set internally by SnapshotSystem` as the comment is from before it was moved.

Reply:
ok, so when we call this function we know we want a Despawn command but we don't know yet what tick it is associated with because we haven't written it yet and are not sure which tick it will be written on?

I guess I always thought the associated tick would be the server tick on which it occurred not when it was written, which I think we would know at this point.

Reply:
The way it is handled, "on which it occurred" _is_ "when it was written". I just didn't want `NetworkObject` to do it because  `NetworkObject` doesn't need to care about time.

```
        internal void Spawn(SnapshotSpawnCommand command)
        {
            command.TickWritten = (ushort)m_CurrentTick;
```
this is called when the spawn/despawn arrives in Snapshot. Not when the command is sent. So, yeah, the original comment was very misleading.

`// will be set internally by SnapshotSystem` is the new comment. Please let me know if it is still unclear